fast_llm/models/ssm/config.py

```diff
@@ -0,0 +1,145 @@
+        # Mamba-specific dimensions
+        tensor_space.add_tensor_dim(TensorDim(SSMDimNames.d_inner, d_inner))
+        tensor_space.add_tensor_dim(TensorDim(SSMDimNames.d_state, self.ssm.state_size))
+        tensor_space.add_tensor_dim(TensorDim(SSMDimNames.dt_rank, mamba_dt_rank))
+        tensor_space.add_tensor_dim(TensorDim(SSMDimNames.d_x_proj, mamba_dt_rank + self.ssm.state_size * 2))
+        tensor_space.add_tensor_dim(TensorDim(SSMDimNames.d_conv_kernel, self.ssm.conv_kernel_dimension))
+
+        if self.ssm.use_mamba2:
+            # as per https://github.com/cartesia-ai/edge/blob/a0e121ebed3d2324c6d762b0e211a08d62583681/cartesia-pytorch/cartesia_pytorch/Llamba/mixers/discrete_mamba2.py#L66C3-L66C4
+            headdim = d_inner // self.ssm.n_v_heads
+            Assert.eq(self.ssm.n_v_heads, d_inner // headdim)
+            Assert.eq(d_inner % headdim, 0)
+            Assert.eq(self.ssm.n_v_heads % self.ssm.n_qk_heads, 0)
+
+            conv_dim = d_inner + 2 * self.ssm.n_qk_heads * self.ssm.state_size
+            inner_proj_dim = 2 * d_inner + 2 * self.ssm.n_qk_heads * self.ssm.state_size + self.ssm.n_v_heads
+
+            tensor_space.add_tensor_dim(TensorDim(SSMDimNames.headdim, headdim))
+            tensor_space.add_tensor_dim(TensorDim(SSMDimNames.n_qk_heads, self.ssm.n_qk_heads))
+            tensor_space.add_tensor_dim(TensorDim(SSMDimNames.n_v_heads, self.ssm.n_v_heads))
+            tensor_space.add_tensor_dim(TensorDim(SSMDimNames.d_inner_proj, inner_proj_dim))
+            tensor_space.add_tensor_dim(TensorDim(SSMDimNames.d_conv, conv_dim))
+        else:
+
+            tensor_space.add_tensor_dim(TensorDim(SSMDimNames.d_inner_proj, d_inner * 2))
+
+    @classmethod
+    def _from_dict(
+        cls,
+        default: dict[str, typing.Any],
+        strict: bool = True,
+        flat: bool = False,
+    ) -> typing.Self:
+        # TODO v0.3: Remove backward compatibility fix
+        if "match_megatron" in default:
+            assert "use_megatron_initialization" not in default
+            default["use_megatron_initialization"] = default.pop("match_megatron")
+        if "layer_norm_impl" in default:
+            assert "normalization_implementation" not in default
+            default["normalization_implementation"] = default.pop("layer_norm_impl")
+        if "fused_mlp" in default:
+            del default["fused_mlp"]
+        return super()._from_dict(default, strict, flat)
+
+    def __post_init__(self):
+        super().__post_init__()
+        if len(self.block_pattern) == 0:
+            logger.warning("No block pattern provided, using default pattern of Transformer blocks.")
+            self.block_pattern = ["t"] * self.transformer.num_layers
+
+    def _validate(self):
+        super()._validate()
+
+
+@config_class()
+class HybridModelConfig(FastLLMModelConfig):
+    _abstract = False
+    model_name: typing.ClassVar[str] = "hybrid_ssm"
```

Comment:
Is this intended to generalize to other Hybrid SSM model? If not we probably want to name it mamba